arch/src/aarch64/layout.rs

```diff
@@ -1,6 +1,59 @@
+//          |                                           | Hole or DRAM      |
+//          |                                           |                   |
+//512GB     +                   +                       +-------------------+
+//          |                                           |       Mapped      |
+//          |                                           |       I/O         |
+//          ~                   ~                       ~                   ~
+//          |                                           |                   |
+//256GB     +                   +                       +-------------------+
+//          |                                           |       Reserved    |
+//          ~                   ~                       ~                   ~
+//          |                                           |                   |
+//64GB      +                   +-----------------------+-------------------+   <- 36-bit
+//          |                   |                   DRAM                    |
+//          ~                   ~                   ~                       ~
+//          |                   |                                           |
+//          |                   |                                           |
+//34GB      +                   +-----------------------+-------------------+
+//          |                   |                  Hole or DRAM             |
+//32GB      +                   +-----------------------+-------------------+
+//          |                   |                   Mapped I/O              |
+//          ~                   ~                       ~                   ~
+//          |                   |                                           |
+//16GB      +                   +-----------------------+-------------------+
+//          |                   |                   Reserved                |
+//          ~                   ~                       ~                   ~
+//4GB       +-------------------+-----------------------+-------------------+   <- 32-bit
+//          |           2GB of DRAM                                         |
+//          |                                                               |
+//2GB       +-------------------+-----------------------+-------------------+
+//          |                           Mapped I/O                          |
+//1GB       +-------------------+-----------------------+-------------------+
+//          |                          ROM & RAM & I/O                      |
+//0GB       +-------------------+-----------------------+-------------------+   0
+//              - 32-bit -              - 36-bit -              - 40-bit -
+//
+// Taken from (http://infocenter.arm.com/help/topic/com.arm.doc.den0001c/DEN0001C_principles_of_arm_memory_maps.pdf).
+
+/// Start of RAM on 64 bit ARM.
+pub const DRAM_MEM_START: usize = 0x80000000; // 2 GB.
```

Comment:
```suggestion
pub const DRAM_MEM_START: usize = 0x8000_0000; // 2 GB.
```

Reply:
now I am wondering why the **clippy** test successfully passed...

Reply:
Maybe clippy doesn't run the arm target code because it isn't run with a arm target.
I think something like:
```cargo clippy --target=arm_something_something``` is required in this case.

Reply:
I will try that but it still does not make sense since our CI uses `cargo clippy --all-targets --all-features`. One would be thinking that `--all-targets` would actually mean all targets :laughing: